src/compiler/parser/expression.rs

```diff
@@ -347,6 +347,42 @@ impl<'a> Parser<'a> {
         })
     }
 
+    pub(super) fn cast(&self, stream: &mut TokenStream) -> ParserResult<Expression<ParserContext>> {
+        // Parser unary expressions
+        let mut msg = None;
+        let (event, result) = self.new_event(Span::zero()).and_then(|| {
+            match self.negate(stream)? {
+                Some(exp) => {
+                    // Check of the `as` operator
+                    match stream.next_if(&Lex::As) {
+                        Some(as_tok) => {
+                            msg = Some("cast");
+                            // If found, then parse a type expression
+                            let (ty, ty_ctx) =
+                                self.consume_type(stream)?.ok_or(CompilerError::new(
+                                    exp.context().join(as_tok.to_ctx()).span(),
+                                    ParserError::InvalidCastTarget,
+                                ))?;
```

Comment:
Major design problem here is that I don't have a `Type Expression` concept in the Bramble compiler or syntax. Which leads to this clumsy stuff and prevents the use of the parsing combinators defined elsewhere.  I should add a Type Expression to the parser and help clean this code up (that's out side the scope of this PR tho)

Reply:
This falls outside the scope of this PR but worth looking into in the next feature that allows for refactorings of the IR model.
 
Possibly this would work well with the CFG implementation.